ts/components/ui/ListItemNav.tsx

```diff
@@ -0,0 +1,207 @@
+import * as React from "react";
+import { useCallback } from "react";
+import {
+  View,
+  StyleSheet,
+  Pressable,
+  GestureResponderEvent,
+  Text
+} from "react-native";
+import Animated, {
+  useSharedValue,
+  useAnimatedStyle,
+  withSpring,
+  useDerivedValue,
+  interpolate,
+  Extrapolate,
+  interpolateColor
+} from "react-native-reanimated";
+import { Icon, IOIcons } from "../core/icons";
+import {
+  IOListItemStyles,
+  IOListItemVisualParams,
+  IOStyles
+} from "../core/variables/IOStyles";
+import { IOSpringValues, IOScaleValues } from "../core/variables/IOAnimations";
+import { LabelSmall } from "../core/typography/LabelSmall";
+import { IOColors, hexToRgba, useIOTheme } from "../core/variables/IOColors";
+import { WithTestID } from "../../types/WithTestID";
+import { useIOSelector } from "../../store/hooks";
+import { makeFontStyleObject } from "../core/fonts";
+import { isDesignSystemEnabledSelector } from "../../store/reducers/persistedPreferences";
+import { NewH6 } from "../core/typography/NewH6";
+import { Body } from "../core/typography/Body";
+
+export type ListItemNav = WithTestID<{
+  value: string;
+  description?: string;
```

Comment:
What do you think about having a description of type `string | React.ReactNode`? In some cases we need to show a skeleton for the loading state of the item, could be useful to pass a component instead of a string.